test/coverlet.collector.tests/CoverletCoverageDataCollectorTests.cs

```diff
@@ -42,6 +44,36 @@ public CoverletCoverageDataCollectorTests()
             _context = new DataCollectionEnvironmentContext(_dataCollectionContext);
             _mockCoverageWrapper = new Mock<ICoverageWrapper>();
             _mockCountDownEventFactory = new Mock<ICountDownEventFactory>();
+
+            var serviceCollection = new ServiceCollection();
+            serviceCollection.AddTransient<IRetryHelper, RetryHelper>();
+            serviceCollection.AddTransient<IProcessExitHandler, ProcessExitHandler>();
+            serviceCollection.AddTransient<IFileSystem, FileSystem>();
+            serviceCollection.AddSingleton<IInstrumentationHelper, InstrumentationHelper>();
+            serviceCollection.AddTransient<ILogger, TestCoverletLogger>();
+            _serviceProvider = serviceCollection.BuildServiceProvider();
+        }
+
+        class TestCoverletLogger : ILogger {
```

Comment:
Can we create a `Mock<ILogger>().Object` in ctor and use it...less code

Reply:
Done